datalad/dochelpers.py

```diff
@@ -372,3 +374,10 @@ def exc_str(exc=None, limit=None, include_str=True):
         # https://docs.python.org/2/library/sys.html#sys.exc_info
         del tb
     return out
+
+
+def exc_str(exc=None, limit=None, include_str=True):
+    """Temporary test shim, for finding issues re refactoring for
+    using CapturedExcpetion instead."""
```

Comment:
Maybe add a note which condition must be fulfilled in order for this shim to be removed.

Reply:
Yeah. If we agree on this PR, I'll put "plans" into an issue and reference it here.

Reply:
as I have mentioned, this PR is IMHO a composition of two largely orthogonal changes:

- encapsulation of exception information in a handy `CapturedExcpetion` (just noted a typo -- should be `CapturedException`).  `exc_str` could just use it to provide formatting (thus no invocations or even logging output needs to change), and `return` records have its instance and add rendering of it etc
- changing output of `exc_str` used throughout codebase in logging to be more contextual (if there is desire/need; and thus may be more silent in some spots)

I think the first one would be easier to review if not clumped together with the 2nd one which might need discussion on its own.

Reply:
@yarikoptic The aim is not to change output of `exc_str`, but move decisions about output changes elsewhere. Please also see #5716 - I think the spots where `exc_str` is used are the wrong spots to make a decision about formatting. The introduction of this class is supposed to help change that. I could do it all at once to make that clear, but that'll be even harder to review. I made this PR to not have it all in one go. Separating further and only having `exc_str` account for a config or something is just inventing yet another intermediate step. I think that'll make it more likely to not get it finished above anything else.
The need for being more silent is discussed in #4478.